core/drivers/crypto/caam/utils/utils_mem.c

```diff
@@ -230,6 +231,35 @@ enum caam_status caam_cpy_block_src(struct caamblock *block,
 	return ret;
 }
 
+enum caam_status caam_cpy_buf_src(struct caambuf *dst, uint8_t *src_data,
+				  size_t src_length)
+{
+	enum caam_status ret = CAAM_FAILURE;
+
+	if (!src_data || !dst)
+		return CAAM_FAILURE;
+
+	if (!src_length)
+		return CAAM_NO_ERROR;
+
+	if (!dst->data) {
+		/* Allocate the destination buffer */
+		ret = caam_alloc_align_buf(dst, src_length);
+		if (ret != CAAM_NO_ERROR) {
+			MEM_TRACE("Allocation buffer error");
+			return ret;
+		}
+	}
+
+	/* Do the copy */
+	memcpy(dst->data, src_data, dst->length);
```

Comment:
What if `dst->length` != `src_length`?

Reply:
It seems a bit fragile, but perhaps not a big problem.
How about adding `assert(dst->length == src_length);` to make it clear that `struct caambuf`'s allocated elsewhere mustn't be used here?

Reply:
Sorry @jenswi-linaro, I didn't get understand your comment. 
`struct caambuf` is allocated here and is also used here.

Reply:
The first time this function is called on n zero-initialized `struct caambuf *`, it will behave as expected and allocate a buffer of sufficient size. All subsequent calls must be made with a `src_length` parameter equal to the initial value. This seems fragile. An assertion that the `src_length` parameter is correct would help catch eventual errors and also make it more obvious that the function is implemented like this on purpose.

Reply:
Got it, Thanks @jenswi-linaro for explanation.
Will add the assert.